crates/re_types/definitions/rerun/datatypes/mat3x3.fbs

```diff
@@ -7,14 +7,46 @@ namespace rerun.datatypes;
 
 // ---
 
-/// A 3x3 column-major Matrix.
+/// A 3x3 Matrix.
+///
+/// Matrices in Rerun are stored as flat list of coefficients in column-major order:
+/// ```text
+///             column 0       column 1       column 2
+///        -------------------------------------------------
+/// row 0 | flat_columns[0] flat_columns[3] flat_columns[6]
+/// row 1 | flat_columns[1] flat_columns[4] flat_columns[7]
+/// row 2 | flat_columns[2] flat_columns[5] flat_columns[8]
+/// ```
+///
+/// \py However, construction is done from a list of rows, which follows NumPy's convention:
+/// \py ```python
+/// \py np.testing.assert_array_equal(
+/// \py     rr.dt.Mat3x3([1, 2, 3, 4, 5, 6, 7, 8, 9]).flat_columns, np.array([1, 4, 7, 2, 5, 8, 3, 6, 9], dtype=np.float32)
+/// \py )
+/// \py np.testing.assert_array_equal(
+/// \py     rr.dt.Mat3x3([[1, 2, 3], [4, 5, 6], [7, 8, 9]]).flat_columns,
+/// \py     np.array([1, 4, 7, 2, 5, 8, 3, 6, 9], dtype=np.float32),
+/// \py )
+/// \py ```
+/// \py If you want to construct a matrix from a list of columns instead, use the named `columns` parameter:
+/// \py ```python
+/// \py np.testing.assert_array_equal(
+/// \py     rr.dt.Mat3x3(columns=[1, 2, 3, 4, 5, 6, 7, 8, 9]).flat_columns,
```

Comment:
Is there a good reason to support this?

I would prefer if the argument to `colums` or `rows` required a shape(3,3).

Reply:
@nikolausWest and I discussed this a bit back and forth. We almost went with not supporting it, but it makes it easier to have no argument being equivalent to `rows`. In turn supporting it on `Mat3x3(my_array)` is nice because it means loading a raw list from a file "just works" for the most part